manifests/init.pp

```diff
@@ -1,60 +1,424 @@
 # == Class: sssd
 #
 # This class allows you to install and configure SSSD.
-#
 # It will forcefully disable nscd which consequently prevents you from using an
 # nscd module at the same time, which is the correct behavior.  ---
 #
+# == Authors
+#
+# * Trevor Vaughan <mailto:tvaughan@onyxpoint.com>
+#
 # == Parameters:
 #
-# @param pki
-#   * If 'simp', include SIMP's pki module and use pki::copy to manage
-#     application certs in /etc/pki/simp_apps/sssd/x509
-#   * If true, do *not* include SIMP's pki module, but still use pki::copy
-#     to manage certs in /etc/pki/simp_apps/sssd/x509
-#   * If false, do not include SIMP's pki module and do not use pki::copy
-#     to manage certs.  You will need to appropriately assign a subset of:
-#     * app_pki_dir
-#     * app_pki_key
-#     * app_pki_cert
-#     * app_pki_ca
-#     * app_pki_ca_dir
-#
-# @param app_pki_external_source
-#   * If pki = 'simp' or true, this is the directory from which certs will be
-#     copied, via pki::copy.  Defaults to /etc/pki/simp/x509.
-#
-#   * If pki = false, this variable has no effect.
-#
-# @param app_pki_dir
-#   This variable controls the basepath of $app_pki_key, $app_pki_cert,
-#   $app_pki_ca, $app_pki_ca_dir, and $app_pki_crl.
-#   It defaults to /etc/pki/simp_apps/sssd/x509.
+# === Installation / Configuration Parameters ===
+# The following parameters allow you to control the installation and
+# configuration of the sssd package.
 #
-# == Authors
+# @param client_package_name [Optional[String]]
```

Comment:
Looks like Puppet-strings is just fine with having the types there when I run it, and also this is the format that Puppet is using in their ntp module (although theirs is much less pretty). 

Regarding default values: at this point the module only supports RedHat, so the defaults are all RedHat defaults. That being said, I don't see why you couldn't just add appropriate data for future operating systems, and add a line to the headers that specifies the default for that OS. Puppet doesn't seem to have a clear position on what the right way to document case-by-case defaults is as of now.